### ISSUE TITLE
Detection

### ISSUE THREAD

**Issue by NicolasSiver** [video-dev/can-autoplay#2]
I'm looking into detection, and have couple doubts:

1. Should we use `playing` event? https://github.com/video-dev/can-autoplay/blob/master/lib/index.js#L31

2. I think it's possible to have video outside of DOM tree - https://github.com/video-dev/can-autoplay/blob/master/lib/index.js#L61

**Comment by caiogondim** [comment]
1. I'm waiting for `canplay` to fire a play. Do you mean if we should wait for `playing` after a `play` attempt?

2. I tested in Chrome and Safari and it indeed runs outside of DOM. But that would make testing harder. Not sure how to properly test this way. https://github.com/video-dev/can-autoplay/blob/master/test/index.js#L38

**Comment by NicolasSiver** [comment]
From my experience, `playing` event what is really matters. Where `canplay` and `canplaythrought` is an API "sugar". For example, `canplay` will be triggered in Microsoft Edge only once in the beginning, but never for a "resume".

**Comment by caiogondim** [comment]
Sure. But I can't call a `play()` before a `canplay` was sent, I guess...
I can change for a `play()` after `canplay` and listen for `playing` with a timeout.

**Comment by NicolasSiver** [comment]
You can. You just asking video tag to play whatever you have provided. You will see a sequence like this:

- `play`
- metadata related + can play flags
- `playing`

I think we should experiment `playing` vs `canplay`.

**Comment by caiogondim** [comment]
Sure. Want me to push a PR or you want to do it and get more familiar with the code (even though it's super small)?

**Comment by NicolasSiver** [comment]
I think, I would like to play with a library to find out, how it works in action. For now, I just looked at the code.

**Comment by caiogondim** [comment]
Please let me know about any feedbacks. 👍

**Comment by gkatsev** [comment]
I'm not sure we want to switch away from `canplay`, the video element is recreated there each time and the promise we get from `play` is the recommended approach to know whether autoplay was rejected or not.

Though, I'm not sure we even necessarily need to wait for the `canplay` event. Just calling play is probably sufficient, especially since I think iOS only triggers `canplay` after you hit play because of how it disables preload by default.

**Comment by NicolasSiver** [comment]
After some code read, actually, `playing` does not make sense. For the implementation, it's enough just `Promise` API from `play()` method.

I'm going to do PR this week. Current work is here: https://github.com/NicolasSiver/can-autoplay/tree/change-api-v2

I need to update tests. It will not be hard since I see a lot of patching and mocking happening.

**Comment by NicolasSiver** [comment]
Ok. It does not work.
I see an inconsistency in the detection.

[can-autoplay-test-video.zip](https://github.com/video-dev/can-autoplay/files/1504216/can-autoplay-test-video.zip)

1. More or less reliable cases for audio and video, where a muted video is the most inconsistent.
2. When you block all auto-playback, muted audio is always successful.

**Comment by NicolasSiver** [comment]
I found the issue. I think it's audio API in the Safari browser. I think they are using internally single "pipe" for the detection. It means, if you run multiple media at the same time they will interfere. So I made an experimental change, where I run only one test, and where I run all test as a queue with a 1-second gap. Everything works and it's predictable. It means, in the real world, checks for `muted` content could be unreliable (if you have multiple videos, picture-in-picture, video editors, etc.)
